File: 0-1 Knapsack Problem/Knapsack.java
/* A Naive recursive implementation of 0-1 Knapsack problem */
public class Knapsack 
{ 
  
    // A utility function that returns maximum of two integers 
     static int max(int a, int b) { return (a > b)? a : b; } 
       
     // Returns the maximum value that can be put in a knapsack of capacity W 
     static int knapSack(int W, int wt[], int val[], int n) 
     { 
        // Base Case 
    if (n == 0 || W == 0) 
        return 0; 
       
    // If weight of the nth item is more than Knapsack capacity W, then 
    // this item cannot be included in the optimal solution 
    if (wt[n-1] > W) 
       return knapSack(W, wt, val, n-1); 
       
    // Return the maximum of two cases:  
    // (1) nth item included  
    // (2) not included 
    else return max( val[n-1] + knapSack(W-wt[n-1], wt, val, n-1), 
                     knapSack(W, wt, val, n-1) 
                      ); 
      } 
  
    
   // Driver program to test above function 
   public static void main(String args[]) 
   { 
        int val[] = new int[]{60, 100, 120}; 
        int wt[] = new int[]{10, 20, 30}; 
    int  W = 50; 
    int n = val.length; 
    System.out.println(knapSack(W, wt, val, n)); 
    } 
} 

File: 0-1 Knapsack Problem/Knapsack.py
#Python implementation of Knapsack problem
#Using a recursive approach

def knapsack(W, wt , v, n):
    if n==0 or W==0:
        return 0
    if (wt[n-1]>W):
        return knapsack(W,wt,v,n-1)
    else:
        return max(v[n-1] + knapsack(W-wt[n-1], wt, v, n-1), knapsack(W,wt,v,n-1))
    
    
    
'''
Example:
    
v = [10,20,30]

wt = [20,10,12]

W = 60

n = len(v)

print(knapsack(W,wt,v,n))
60

v = [10,20,10]

wt = [20,10,30]

W = 50

n = len(v)

print(knapsack(W,wt,v,n))
30'''

